### PR TITLE
Other id ref

### PR DESCRIPTION
**Pull request by jsstevenson-tmp**
Combining other_id references + the VOD update so we only have to merge the issue-90 stuff once

### CODE REVIEW

File: therapy/etl/chemidplus.py [hunk 1]
```diff
@@ -164,21 +164,38 @@ def _load_record(self, batch: BatchWriter, record: Dict):
...
+        concept_id_ref = record['concept_id'].lower()
+        aliases = record.get('aliases')
+        if aliases:
+            for alias in {a.casefold() for a in aliases}:
+                batch.put_item(Item={
+                    'label_and_type': f'{alias}##alias',
+                    'concept_id': concept_id_ref,
+                    'src_name': SourceName.CHEMIDPLUS.value,
+                })
+        elif 'aliases' in record:
```
Comment (korikuzma) [inline therapy/etl/chemidplus.py:176]: Aliases always exists, see [here](https://github.com/cancervariants/therapy-normalization/blob/427a4228080f261b682f5358e59243a1f1c6ece3/therapy/etl/chemidplus.py#L141). 
```suggestion
        else:
```

File: therapy/etl/chemidplus.py [hunk 2]
```diff
@@ -164,21 +164,38 @@ def _load_record(self, batch: BatchWriter, record: Dict):
...
+            del record['aliases']
+        if record.get('label'):
             batch.put_item(Item={
                 'label_and_type': f'{record["label"].lower()}##label',
-                'concept_id': concept_id_l,
-                'src_name': SourceName.CHEMIDPLUS.value
+                'concept_id': concept_id_ref,
+                'src_name': SourceName.CHEMIDPLUS.value,
             })
+        elif 'label' in record:
```
Comment (korikuzma) [inline therapy/etl/chemidplus.py:184]: Label is also always present
```suggestion
        else:
```

File: therapy/etl/chemidplus.py [hunk 3]
```diff
@@ -164,21 +164,38 @@ def _load_record(self, batch: BatchWriter, record: Dict):
...
+            del record['label']
+        other_ids = record.get('other_identifiers')
+        if other_ids:
+            for other_id in {i.casefold() for i in other_ids}:
+                batch.put_item(Item={
+                    'label_and_type': f'{other_id}##other_id',
+                    'concept_id': concept_id_ref,
+                    'src_name': SourceName.CHEMIDPLUS,
+                })
+        elif 'other_identifiers' in record:
```
Comment (korikuzma) [inline therapy/etl/chemidplus.py:194]:
```suggestion
        else:
```

File: therapy/query.py [hunk 4]
```diff
@@ -391,57 +346,124 @@ def _record_order(self, record: Dict) -> (int, str):
...
+        """Format received DB record as VOD and update response object.
+
+        :param Dict response: in-progress response object
+        :param Dict record: record as stored in DB
+        :param str query: query string from user request
+        :param MatchType match_type: type of match achieved
+        :return: completed response object ready to return to user
+        """
+        vod = {
+            'id': f'normalize:{quote(query)}',
```
Comment (korikuzma) [inline therapy/query.py:362]:
```suggestion
            'id': f'normalize.therapy:{quote(query)}',
```

File: therapy/schemas.py [hunk 5]
```diff
@@ -315,240 +316,67 @@ def schema_extra(schema: Dict[str, Any],
...
-                }
-            }
+            schema['example'] = {}  # TODO
 
 
-class NormalizationService(BaseModel):
-    """Response containing one or more merged records and relevant source
-    data.
-    """
+class Service(BaseModel):
```
Comment (korikuzma) [inline therapy/schemas.py:364]: Could we rename this to SearchService or something with Search in it?

**Review by korikuzma — CHANGES_REQUESTED** [review]

File: therapy/etl/chemidplus.py (hunk 1 above)
Comment (jsstevenson-tmp) [reply therapy/etl/chemidplus.py:176]: 👍

File: therapy/query.py [hunk 6]
```diff
@@ -391,57 +346,124 @@ def _record_order(self, record: Dict) -> (int, str):
...
+        """Format received DB record as VOD and update response object.
+
+        :param Dict response: in-progress response object
+        :param Dict record: record as stored in DB
+        :param str query: query string from user request
+        :param MatchType match_type: type of match achieved
+        :return: completed response object ready to return to user
+        """
+        vod = {
+            'id': f'normalize.disease:{quote(query)}',
```
Comment (korikuzma) [inline therapy/query.py:362]: Am I hallucinating? Are we in disease or therapy. I think it should be therapy.

**Review by korikuzma — APPROVED** [review]

**Pull request merged**